src/searchlog.c

```diff
@@ -182,75 +220,78 @@ int displayPartials(char *suggested_call) {
      * be aware of the problem of marking it dupe only for a complete
      * match.
      */
-    for (k = 0; k < srch_index; k++) {
-	if (strlen(hiscall) >= 2 && strlen(searchresult[k]) > 2) {
-	    if (strstr(searchresult[k], hiscall) != NULL) {
-		printres[0] = '\0';
-		strncat(printres, searchresult[k] + 29, 12);
-
-		/* cut string just at first space after call */
-		loc = strchr(printres, ' ');
-		if (loc)
-		    *loc = '\0';
-
-		if (dupe == ISDUPE) {
-		    attrset(COLOR_PAIR(C_DUPE));
-		} else {
-		    attron(modify_attr(COLOR_PAIR(C_BORDER) |
-				       A_STANDOUT));
-		}
-		mvprintw(xwin + l, ywin + j, "%s ", printres);
-		attron(COLOR_PAIR(C_LOG) | A_STANDOUT);
+    if (dupe == ISDUPE) {
+	attrset(COLOR_PAIR(C_DUPE));
+    } else {
+	attron(modify_attr(COLOR_PAIR(C_BORDER) | A_STANDOUT));
+    }
 
-		attroff(A_BOLD);
+    GHashTable *callset = g_hash_table_new(g_str_hash, g_str_equal);
+    int full = 0;
 
-		j += (strlen(printres) + 1);
+    for (k = 0; k < srch_index && !full; k++) {
+	if (strlen(searchresult[k]) <= 2) {
+	    continue;   // too short (unlikely)
+	}
+	if (strstr(searchresult[k], hiscall) == NULL) {
+	    continue;   // not matching
+	}
 
-		if (j >= 30) {
-		    l++;
-		    j = 0;
+	printres[0] = '\0';
+	strncat(printres, searchresult[k] + 29, 12);
 
-		}
-		if (l > 4)
-		    break;
-	    }
-	}
-    }
+	/* cut string just at first space after call */
+	loc = strchr(printres, ' ');
+	if (loc)
+	    *loc = '\0';
 
-    if (strcmp(hiscall, printres) != 0) {
+	full = show_partial(&row, &col, printres, callset,
+			    &suggested, suggested_call);
 
-	/* and now check callmaster database */
-	for (k = 0; k < callmaster->len; k++) {
+    }
 
-	    if (strstr(CALLMASTERARRAY(k), hiscall) != NULL) {
+    attron(COLOR_PAIR(C_LOG) | A_STANDOUT);
+    attroff(A_BOLD);
 
-		attron(modify_attr(COLOR_PAIR(C_LOG) | A_STANDOUT));
+    // no callmaster check if exact match was found in the log
```

Comment:
What happens if call found in log is a substring of one in callmaster (or log), e.g. K3A and K3AL?

Reply:
I tried to keep search functionality unchanged, so let's see.
Is this the setup?
- in log: K3A
- in callmaster: K3AL
- entered: K3A

In this case current code gives only K3A and as dupe. Same with the updated code.

Do you mean it should also list K3AL as an option?

Reply:
Yes, but it should work in any combination of log and callmaster. Maybe you already worked K3A. As you use the partial function to look up calls which may fit to that partial string K3AL from callmaster is a valid possibility.

It seems that the old implementation is wrong here.

Reply:
In the current code this line prevented further search
```
    if (strcmp(hiscall, printres) != 0) {

        /* and now check callmaster database */
```

We could simply ignore this check and do all searches even if call was in the log already.
Makes sense. The display would be then like this (K3AL comes first as I added it extra, so my callmaster file is not sorted)
![image](https://user-images.githubusercontent.com/15141948/69914974-36ef2880-144a-11ea-8e46-007cc33921bb.png)


In this case USEPARTIALS wouldn't trigger, but that's OK as the call has been entered anyway.

I'll change the display code a bit to do a tighter checking that would allow K3ATO fit on the first line and thus showing a bit more results.